### ISSUE TITLE
Material icons extended only being published for desktop

### ISSUE THREAD

**Issue by luca992** [JetBrains/compose-jb#2106]
I was having trouble using the extended material icons in my common code. It turns out it is only being built/published for desktop.

You can verify here:
https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/

Material icons core does not have this issue.

**Comment by igordmn** [comment]
We only publish it for desktop. For android it should be redirected to the Jetpack Compose published version.

I just checked Compose 1.1.1, and it works fine, if we add icons this way (in MPP-style project):
```
kotlin {
    ...
    sourceSets {
        named("commonMain") {
            dependencies {
                ...
                api(compose.materialIconsExtended)
            }
        }
    }
}
```

and use `Icons.Filled.BlurOn`

**Comment by igordmn** [comment]
Redirecting is described in this file:
https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/material-icons-extended/1.1.1/material-icons-extended-1.1.1.module (look for androidx.compose.material)

**Comment by luca992** [comment]
Okay, if that's on purpose, why? The basic icon set is published for all targets as I mentioned. Shouldn't the extended icons be published for all targets as well?

**Comment by luca992** [comment]
> We only publish it for desktop. For android it should be redirected to the Jetpack Compose published version.
> 
> I just checked Compose 1.1.1, and it works fine, if we add icons this way (in MPP-style project):
> 
> ```
> kotlin {
>     ...
>     sourceSets {
>         named("commonMain") {
>             dependencies {
>                 ...
>                 api(compose.materialIconsExtended)
>             }
>         }
>     }
> }
> ```
> 
> and use `Icons.Filled.BlurOn`

Sorry, what I reported was somewhat incorrect. 

The project will build with `implementation(compose.materialIconsExtended)` in `commonMain`. But, Intellij will fail to resolve any extended icon in the `commonMain` source set. Which is just really annoying plus code completion for the extended icons doesn't work.

to work around that issue I added `implementation("org.jetbrains.compose.material:material-icons-extended-desktop:_")` to both jvm and android dependencies and everything works as it should.

However, I have been trying out the ios support on the alpha build and it would be nice to have the icons published for all the targets

**Comment by Antimonit** [comment]
I just ran into the same issue as @luca992.

`Icons.Default.Add` is in the `material-icons-core` package.
`Icons.Default.Remove` is in the `material-icons-extended` package.

* With `api(compose.materialIconsExtended)`
  * `Icons.Default.Remove` is highlighted red and autocomplete for icons from the extended package does not work.
* With `api(compose("org.jetbrains.compose.material:material-icons-extended-desktop"))`
  * `Icons.Default.Remove` is correctly recognized.

In both cases, the app compiles just fine but IntelliJ IDEA highlighting and autocomplete are broken in the first scenario.

**Comment by Thomas-Vos** [comment]
@igordmn I think when these packages (icons extended, material3, etc) are published for all targets that the following issue may be fixed as well. It is about autocomplete not working in common source sets. https://github.com/JetBrains/compose-jb/issues/2032 (see comments). It seems to only affect the packages that are not published for all targets.

**Comment by luca992** [comment]
`1.2.0-alpha01-dev731` is working for me in common and building for jvm and android. Thanks!

**Comment by lucasmeneghin** [comment]
material-icons-extended-js is still missing from https://maven.pkg.jetbrains.space/public/p/compose/dev/org/jetbrains/compose/material/

**Comment by igordmn** [comment]
> material-icons-extended-js is still missing from

Indeed. Thanks for noticing.

Reopening this issue until we fix it for js.

**Comment by luca992** [comment]
is there any eta on js? @igordmn

**Comment by lucasmeneghin** [comment]
can we have this fixed? :(
@igordmn

**Comment by eygraber** [comment]
@igordmn is there anything to do here besides making sure that artifact gets published correctly?

**Comment by igordmn** [comment]
> @igordmn is there anything to do here besides making sure that artifact gets published correctly?

Probably not, but I can't tell for sure. [This](https://github.com/JetBrains/compose-jb/blob/7e574a07e23c8766770227f767046d95c627b237/compose/build.gradle.kts#L87) task should publish icons for all platforms. This issue isn't in our near plans, but if someone will make a PR with the fix, ping me in the PR, and mention this issue, I will review and merge.

**Comment by eygraber** [comment]
I'll see if I can find anything.

Would it be possible to manually build and deploy the artifacts for existing versions (1.2.1, 1.3.0-beta, etc...)?

**Comment by igordmn** [comment]
> possible to manually build 

No, it requires much more efforts comparing to just publishing for the next versions (our CI isn't configured for this)

**Comment by eygraber** [comment]
How does CI run `publishComposeJbExtendedIcons`?

Without seeing that, my best guess would be that [`compose.platforms`](https://github.com/JetBrains/compose-jb/blob/7e574a07e23c8766770227f767046d95c627b237/compose/buildSrc/src/main/kotlin/ComposeProperties.kt#L14) either isn't set or doesn't specify `js` when running [`publishExtendedIcons`](https://github.com/JetBrains/compose-jb/blob/7e574a07e23c8766770227f767046d95c627b237/compose/scripts/publishExtendedIcons)

**Comment by igordmn** [comment]
[Here](https://teamcity.jetbrains.com/admin/editBuildRunners.html?id=buildType:JetBrainsPublicProjects_Compose_PublishFast_1ComposeExtendedIconsPublish) is the CI task (you can create an account on this TeamCity if you don't have one)

![image](https://user-images.githubusercontent.com/5963351/204786132-c1d0220b-3221-4d05-aded-d08e1e09ee2f.png)
![image](https://user-images.githubusercontent.com/5963351/204786358-3d25ec3b-f79b-4a56-a6d5-3bdae6a829e1.png)
![image](https://user-images.githubusercontent.com/5963351/204786230-25e78b1a-99cb-4616-82ff-2f1697b4deb0.png)

**Comment by igordmn** [comment]
I looked on the other task, and found out that for JS we call `publish` tasks separately with an additional argument. Added a new task:

![image](https://user-images.githubusercontent.com/5963351/204787517-954550cb-28af-48ca-86ee-91e0fd87ca30.png)

And ran the new 1.3.0-beta

**Comment by eygraber** [comment]
Awesome thank you!

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2106">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
